### PR TITLE
Feature/migration

### PR DESCRIPTION
**Pull request by andhikayuana**
### Better local data migration : 

1. write `sql` script with naming like **qiscus.db_from_oldVersion_to_newVersion.sql**
   exmaple : **qiscus.db_from_14_to_15.sql**
2. write your sql script inside that file : 

   example : 

```sql
ALTER TABLE comments ADD COLUMN TESTING TEXT;
UPDATE comments SET TESTING = sender_email;
```

3. place above file inside `assets` directory

4. increase DB version `com.qiscus.sdk.data.local.QiscusDb.DATABASE_VERSION`

Voila :smile: 

### Test migration 

1. [SQL Script for migration](https://github.com/andhikayuana/qiscus-sdk-android/blob/feature/migration/chat/src/main/assets/qiscus.db_from_14_to_15.sql)
2. [Increase DB Version](https://github.com/andhikayuana/qiscus-sdk-android/blob/1a1ade5123913ed3a872cddf4393915d842497cc/chat/src/main/java/com/qiscus/sdk/data/local/QiscusDb.java#L33)

### CODE REVIEW

**Comment by andhikayuana** [comment]
bang @zetbaitsu ini udah ada `examplenya` juga, eh ini examplenya tak ganti ke bikik index ke field aja sekalian :smile:

File: chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java [hunk 1]
```diff
@@ -17,13 +17,25 @@
...
+import com.qiscus.sdk.util.QiscusLogger;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
 
 class QiscusDbOpenHelper extends SQLiteOpenHelper {
 
+    private Context mContext;
```
Comment (zetbaitsu) [inline chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java]: pak @andhikayuana request dikit ya, namanya ganti jadi context aja pak biar seragam sama yg lain, soalnya kita ga pake prefix m gitu.

File: chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java [hunk 2]
```diff
@@ -43,21 +55,62 @@ public void onCreate(SQLiteDatabase db) {
...
-            db.execSQL("DROP TABLE IF EXISTS " + QiscusDb.RoomTable.TABLE_NAME);
-            db.execSQL("DROP TABLE IF EXISTS " + QiscusDb.MemberTable.TABLE_NAME);
-            db.execSQL("DROP TABLE IF EXISTS " + QiscusDb.RoomMemberTable.TABLE_NAME);
-            db.execSQL("DROP TABLE IF EXISTS " + QiscusDb.CommentTable.TABLE_NAME);
-            db.execSQL("DROP TABLE IF EXISTS " + QiscusDb.FilesTable.TABLE_NAME);
-            db.setTransactionSuccessful();
+            InputStream inputStream = assets.open(migrationName);
+            InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
+            reader = new BufferedReader(inputStreamReader);
+            execSQLScript(db, reader);
```
Comment (zetbaitsu) [inline chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java]: exec sql nya di lock pake transaction ya pak
```
db.beginTransaction();
try {
   ====SQL===
   db.setTransactionSuccessful();
} finally {
   db.endTransaction();
}
```

**Review by zetbaitsu — CHANGES_REQUESTED** [review]

File: chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java (hunk 1 above)
Comment (andhikayuana) [reply chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java]: oke

File: chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java (hunk 2 above)
Comment (andhikayuana) [reply chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java]: noted

File: chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java [hunk 3]
```diff
@@ -43,21 +55,68 @@ public void onCreate(SQLiteDatabase db) {
...
+            String line;
+            StringBuilder statement = new StringBuilder();
+            while ((line = reader.readLine()) != null) {
+                statement.append(line);
+                statement.append(System.getProperty("line.separator"));
+                if (line.endsWith(";")) {
+                    db.execSQL(statement.toString());
+                    statement = new StringBuilder();
+                }
+            }
```
Comment (zetbaitsu) [inline chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java:116]: kurang ini pak ```db.setTransactionSuccessful();```

**Review by zetbaitsu — CHANGES_REQUESTED** [review]

File: chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java (hunk 3 above)
Comment (andhikayuana) [reply chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java:116]: wohiya

Comment (zetbaitsu) [reply chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java:116]: eh pak @andhikayuana ini jangan lupa ya :v

Comment (andhikayuana) [reply chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java:116]: eh sudah pak [sini](https://github.com/qiscus/qiscus-sdk-android/pull/48/commits/c1e756cab8d41a96b9681b8435488849ef6d79d5#diff-863ce0810c62d01a41f2fc33ab598d47R117)

Comment (zetbaitsu) [reply chat/src/main/java/com/qiscus/sdk/data/local/QiscusDbOpenHelper.java:116]: oh udah ya :v

**Review by zetbaitsu — APPROVED** [review]

**Pull request merged**